### DISCUSSION TITLE
journalctl command using invalid syntax for --vacuum-time= argument

### DISCUSSION THREAD

**Discussion by hudson4351** [bleachbit/bleachbit#1411] [Q&A]
When I select the "journald" option (and its sub option "clean") from the main window and then click "Preview", the output is as follows:

```
clean_operation('journald'), options = '['clean']'
journalctl --vacuum-time=1 ?B 

Disk space to be recovered: 0B
Files to be deleted: 0
Special operations: 1
elapsed time: 0 seconds
```

The syntax to the `--vacuum-time=` argument appears to be invalid; consequently BleachBit does not find any journal files to remove. How do I specify a proper number and suffix for BleachBit to use with the journalctl command? This is running Ubuntu 22.04.1 LTS.

**Comment by az0** [comment]
It's in the Aciton.py file here

https://github.com/bleachbit/bleachbit/blob/63f0513723d8bcafc72cd80c193c738bed64703b/bleachbit/Action.py#L477

What should it be?

**Reply by hudson4351** [reply]
Based on the man page it looks like a suffix is required:

```
--vacuum-size=, --vacuum-time=, --vacuum-files=
           Removes the oldest archived journal files until the disk
           space they use falls below the specified size (specified with
           the usual "K", "M", "G" and "T" suffixes), or all archived
           journal files contain no data older than the specified
           timespan (specified with the usual "s", "m", "h", "days",
           "months", "weeks" and "years" suffixes), or no more than the
           specified number of separate journal files remain.
```

source: https://man7.org/linux/man-pages/man1/journalctl.1.html

Could this be made into a user-specified option in the Preferences menu? Maybe a radio button to select one of "size", "time", or "files", and then a combo box to select a valid suffix ("size" and "time" options) or number of files ("files" option)?

**Reply by abitrolly** [reply]
Here is where it is defined.

https://github.com/systemd/systemd/blob/1d87f03a6e62476b996bccaba62af329aa40ba1c/src/journal/journalctl.c#L774-L780

`parse_sec`
https://github.com/systemd/systemd/blob/6f2d367e4971c462c382d607825441bc294b28cd/src/basic/time-util.c#L1101-L1103

**Reply by abitrolly** [reply]
According to tests, the suffix is not required and it will be equal to `1s` - https://github.com/systemd/systemd/blob/aa181897cf189263e2786d8a184918442e52ee1a/src/test/test-time-util.c

**Reply by abitrolly** [reply]
Added PR to document it https://github.com/systemd/systemd/pull/25189

**Comment by abitrolly** [comment]
It looks like the command is not executed. I modified the line to exec `amnothere` and there is no error.

https://github.com/bleachbit/bleachbit/blob/63f0513723d8bcafc72cd80c193c738bed64703b/bleachbit/Action.py#L477

```
clean_operation('journald'), options = '['clean']'
amnothere --vacuum-time=1 0B 
Disk space recovered: 0B
Files deleted: 0
Special operations: 1
```
